securedrop_client/storage.py

```diff
@@ -294,15 +295,37 @@ def find_or_create_user(uuid, username, session):
 
 
 def find_new_messages(session):
-    return session.query(Message).filter_by(is_downloaded=False).all()
+    """
+    Find messages to process. Those messages are those where one of the following
+    conditions is true:
+
+    * The message has not yet been downloaded.
+    * The message has not yet had decryption attempted.
+    * Decryption previously failed on a message.
+    """
+    return session.query(Message).filter(
+        or_(Message.is_downloaded == False,
+            Message.is_decrypted == False,
+            Message.is_decrypted == None)).all()  # noqa E711
 
 
 def find_new_files(session):
     return session.query(File).filter_by(is_downloaded=False).all()
 
 
 def find_new_replies(session):
-    return session.query(Reply).filter_by(is_downloaded=False).all()
+    """
+    Find replies to process. Those replies are those where one of the following
+    conditions is true:
+
+    * The reply has not yet been downloaded.
+    * The reply has not yet had decryption attempted.
+    * Decryption previously failed on a reply.
+    """
+    return session.query(Reply).filter(
+        or_(Reply.is_downloaded == False,
+            Reply.is_decrypted == False,
+            Reply.is_decrypted == None)).all()  # noqa E711
```

Comment:
This needs a colon directly after `noqa` otherwise `flake8` will ignore all errors on that line.